Sources/SWBTestSupport/CoreTestSupport.swift

```diff
@@ -74,33 +74,35 @@ extension Core {
         //
         // If the given environment already contains `EXTERNAL_TOOLCHAINS_DIR` and `TOOLCHAINS`, we're assuming that we do not have to obtain any toolchain information.
         var environment = environment
-        if (try? ProcessInfo.processInfo.hostOperatingSystem()) == .macOS, !(environment.contains("EXTERNAL_TOOLCHAINS_DIR") && environment.contains("TOOLCHAINS")) {
-            let activeDeveloperPath: Path
-            if let developerPath {
-                activeDeveloperPath = developerPath.path
-            } else {
-                activeDeveloperPath = try await Xcode.getActiveDeveloperDirectoryPath()
-            }
-            let defaultToolchainPath = activeDeveloperPath.join("Toolchains/XcodeDefault.xctoolchain")
-
-            if !localFS.exists(defaultToolchainPath.join("usr/metal/current")) {
-                struct MetalToolchainInfo: Decodable {
-                    let buildVersion: String
-                    let status: String
-                    let toolchainIdentifier: String
-                    let toolchainSearchPath: String
+        if configurationDelegate?.loadMetalToolchain == true {
+            if (try? ProcessInfo.processInfo.hostOperatingSystem()) == .macOS, !(environment.contains("EXTERNAL_TOOLCHAINS_DIR") && environment.contains("TOOLCHAINS")) {
+                let activeDeveloperPath: Path
+                if let developerPath {
+                    activeDeveloperPath = developerPath.path
+                } else {
+                    activeDeveloperPath = try await Xcode.getActiveDeveloperDirectoryPath()
                 }
-
-                let result = try await Process.getOutput(url: URL(fileURLWithPath: activeDeveloperPath.join("usr/bin/xcodebuild").str), arguments: ["-showComponent", "metalToolchain", "-json"], environment: ["DEVELOPER_DIR": activeDeveloperPath.str])
-                if result.exitStatus != .exit(0) {
-                    throw StubError.error("xcodebuild failed: \(String(data: result.stdout, encoding: .utf8) ?? "")\n\(String(data: result.stderr, encoding: .utf8) ?? "")")
+                let defaultToolchainPath = activeDeveloperPath.join("Toolchains/XcodeDefault.xctoolchain")
+
+                if !localFS.exists(defaultToolchainPath.join("usr/metal/current")) {
+                    struct MetalToolchainInfo: Decodable {
+                        let buildVersion: String
+                        let status: String
+                        let toolchainIdentifier: String
+                        let toolchainSearchPath: String
+                    }
+
+                    let result = try await Process.getOutput(url: URL(fileURLWithPath: activeDeveloperPath.join("usr/bin/xcodebuild").str), arguments: ["-showComponent", "metalToolchain", "-json"], environment: ["DEVELOPER_DIR": activeDeveloperPath.str])
```

Comment:
This should use `runProcess` instead, since it already wraps `getOutput` with the necessary plumbing to emit a good error message if the command fails.